### PR TITLE
Fix MTE-3383 - for testTabTrayContextMenuCloseTab test

### PR DESCRIPTION
**Pull request by dragosb01**
## :scroll: Tickets
https://mozilla-hub.atlassian.net/browse/MTE-3383

## :bulb: Description
Fixed testTabTrayContextMenuCloseTab flaky test by adding a wait after pressing undo button, making sure the tab gets restored before counting the total number of tabs.

Moved smoke tests from ActivityStreamTest and BookmarksTests suites from test plan1 to test plan4 in order to balance the execution time.

### CODE REVIEW

File: firefox-ios/firefox-ios-tests/Tests/Smoketest1.xctestplan [hunk 1]
```diff
@@ -15,13 +15,17 @@
   "testTargets" : [
     {
       "skippedTests" : [
+        "ActivityStreamTest",
         "ActivityStreamTest\/testContextMenuInLandscape()",
+        "ActivityStreamTest\/testDefaultSites()",
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/Smoketest1.xctestplan:20]: @dragosb01 We are removing tests from the SMoketest 1, are those in other test suites already? I see we have 22 tests in smoketest 1, if we land this PR we will have 15

Comment (dragosb01) [reply firefox-ios/firefox-ios-tests/Tests/Smoketest1.xctestplan:20]: @isabelrios the tests were moved in Smoketest4:
https://addons-testing.bitrise.io/builds/82d437d5-f56d-432b-af9b-34eb52053306/testreport/1b78e3fc-354a-4534-8fbc-1ebda9dcf98e/testsuite/0/testcases?status=failed
The tests are still executed under our smoke plans.

Comment (dragosb01) [reply firefox-ios/firefox-ios-tests/Tests/Smoketest1.xctestplan:20]: As you can see we have 7 more tests on Smoketest4, the difference from 22 to 15 from Smoketest1

Comment (isabelrios) [reply firefox-ios/firefox-ios-tests/Tests/Smoketest1.xctestplan:20]: oh I see yeah, I saw that.. the smoketest 1 takes longer than the others..

**Review by isabelrios — APPROVED** [review]

**Comment by dragosb01** [comment]
@Mergifyio backport release/v131

**Pull request merged**